sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperClient.java

```diff
@@ -175,9 +176,13 @@ private Map<String, List<String>> fuzzyGetConfigListByGroupAndKey(String key, St
     private void fillChildrenInfo(String path, Map<String, List<String>> configMap, String nodeName,
                                   String key) throws InterruptedException, KeeperException {
         List<String> children = this.zkClient.getChildren(path, false);
+        if (children == null || children.isEmpty()) {
+            return;
+        }
+        Pattern groupPattern = Pattern.compile(nodeName);
```

Comment:
cache patterns like the comment in kieclient

Reply:
done